thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs

```diff
@@ -429,21 +452,32 @@ a3corsPolicy = CorsPolicy
 
 -- | Create a user creation request to be sent to the A3 backend. The actual user password is
 -- replaced by a dummy, as A3 doesn't have to know it.
+mkUserCreationRequestForA3 :: ThentosConfig -> UserFormData -> Maybe Client.Request
+mkUserCreationRequestForA3 config user = do
+    let user'  = UserFormData { udName     = udName user,
+                                udEmail    = udEmail user,
+                                udPassword = "dummypass" }
+    mkRequestForA3 config "/principals/users" $ A3UserWithPass user'
+
+-- | Make a POST request to be sent to the A3 backend. Returns 'Nothing' if the 'ThentosConfig'
+-- lacks a correctly configured proxy.
+--
+-- Note that the request is configured to NOT thrown an exception even if the response status code
+-- indicates an error (400 or higher). Properly dealing with error replies is left to the caller.
 --
 -- Since the A3 frontend doesn't know about different services (i.e. never sends a
 -- @X-Thentos-Service@ header), we send the request to the default proxy which should be the A3
 -- backend.
-mkUserCreationRequestForA3 :: ThentosConfig -> UserFormData -> Maybe Client.Request
-mkUserCreationRequestForA3 config user = do
+mkRequestForA3 :: ToJSON a => ThentosConfig -> String -> a -> Maybe Client.Request
+mkRequestForA3 config route dat = do
     defaultProxy <- Tagged <$> config >>. (Proxy :: Proxy '["proxy"])
     let target = extractTargetUrl defaultProxy
-        user'  = UserFormData { udName     = udName user,
-                                udEmail    = udEmail user,
-                                udPassword = "dummypass" }
-    initReq <- Client.parseUrl $ cs $ show target <> "/principals/users"
-    return initReq { Client.method = "POST",
-        Client.requestHeaders = [("Content-Type", "application/json")],
-        Client.requestBody = Client.RequestBodyLBS . Aeson.encode . A3UserWithPass $ user' }
+    initReq <- Client.parseUrl $ cs $ show target <> route
```

Comment:
Functions that invoke an external service are "partial" by default. We could make sure that the service is reachable at startup, but it could always go down later or the network connection could break down. So that can't be helped.